Sources/Datadog/Core/Attributes/UserInfoProvider.swift

```diff
@@ -35,12 +35,24 @@ internal class UserInfoProvider {
     }
 }
 
-/// Information about the user.
-internal struct UserInfo {
-    let id: String?
-    let name: String?
-    let email: String?
-    let extraInfo: [AttributeKey: AttributeValue]
-
-    internal static var empty: UserInfo { UserInfo(id: nil, name: nil, email: nil, extraInfo: [:]) }
+public struct UserInfo {
```

Comment:
This `UserInfo` is part of `Datadog/Core`. Both Tracing and RUM use their own representation of user info:
* [`SpanEvent.UserInfo` for Tracing](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/Tracing/Span/SpanEventEncoder.swift#L68) - it has slightly different signature (`[String: String]` for extra info),
* [`RUMUser` for RUM](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/RUM/DataModels/RUMDataModels.swift#L1557) -it is auto-generated model from RUM schema.

I'm in between:
* 1/ exposing this `Core.UserInfo` only for `LogEvent`,
* 2/ creating `LogEvent.UserInfo` and having mapper.

Going with the 2nd approach would make the architecture aligned with other products for `UserInfo`, but would require duplicating its definition. I personally prefer small duplication over bad abstraction and that might be the case given that we're planning to modularize our SDK soon into separate libraries (Tracing / RUM / Logging).

On the other hand, same thing applies to `NetworkConnectionInfo` and `CarrierInfo` which are both `public` and used from `SpanEvent` and (now) `LogEvent`, but not `RUMEvent` (RUM maps both into a single [`RUMConnectivity`](https://github.com/DataDog/dd-sdk-ios/blob/181359bd655b894041702332c26fa05b2d451990/Sources/Datadog/RUM/DataModels/RUMDataModels.swift#L1469) value). That would need to be tackled somehow later in mentioned modularization.

What's your take on that?

Reply:
I had the exact same thought process, I started by creating an inner `LogEvent.UserInfo` but then I noticed both `NetworkConnectionInfo` and `CarrierInfo` were shared.

For modularization prospect, using contained model definition per feature and use abstractions for shared data would be a good approach IMHO. Using an inner UserInfo would be a step in that direction and facilitate the transition. I will do the change!